server/task_check_scheduler.go

```diff
@@ -179,6 +179,69 @@ func (s *TaskCheckScheduler) Register(taskType string, executor TaskCheckExecuto
 
 // ScheduleCheckIfNeeded schedules a check if needed.
 func (s *TaskCheckScheduler) ScheduleCheckIfNeeded(ctx context.Context, task *api.Task, creatorID int, skipIfAlreadyTerminated bool) (*api.Task, error) {
+	if task.NotBeforeTs != 0 {
+		// All task should pass timing task check.
+		// Since Time is an auto-increment value, a task would generate a timing task check every time before executing.
+		// However, this is an annoying behavior, and we adopt the following logic to ease this problem:
+		// 		1. if no timing check has been schedule yet, schedule one
+		// 		2. if one has been scheduled before:
+		//			a. succeed
+		//				check again, since users are allowed to change this field
+		//			b. failed
+		//     			*** check if it has passed the not_before_ts ***
+		// 					b-1. passed
+		//						schedule a new timing task check
+		// 					b-2. not passed
+		//						do nothing
+		// Notice that if no timing task check has been run before, the value of isTimingTaskCheckPassed would still be false
+		isTimingTaskCheckPassed, err := s.server.passCheck(ctx, s.server, task, api.TaskCheckGeneralEarliestAllowedTime)
+		if err != nil {
+			return nil, err
+		}
+		if isTimingTaskCheckPassed {
```

Comment:
Looks like the block can be simplified.

There are 2 variations.

1. Whether we need to run the check

2. If we need to run the check, whether we need to pass SkipIfAlreadyTerminated

Would be more readable to structrure and comment code in such way.